### PR TITLE
Feature/ef

### PR DESCRIPTION
**Pull request by pavelbannov**
DbManager replaced by EF Core

### CODE REVIEW

File: common/ASC.Core.Common/EF/Model/DbWebstudioSettings.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public static class WebstudioSettingsExtension
+    {
+        public static ModelBuilder AddWebstudioSettings(this ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<DbWebstudioSettings>()
+                .HasKey(c => new { c.TenantId, c.Id, c.UserId });
+
+            //modelBuilder.Entity<DbWebstudioSettings>()
```
Comment (NikolayRechkin) [inline common/ASC.Core.Common/EF/Model/DbWebstudioSettings.cs]: remove comment

File: common/ASC.Core.Common/EF/Model/Mail/Mailbox.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        public int MsgCountLast { get; set; }
+
+        [Column("size_last")]
+        public int SizeLast { get; set; }
+
+        [Column("login_delay")]
+        public int LoginDelay { get; set; }
+
+        [Column("quota_error")]
+        public int QuotaError { get; set; }
```
Comment (NikolayRechkin) [inline common/ASC.Core.Common/EF/Model/Mail/Mailbox.cs]: need boolean data type

File: common/ASC.Core.Common/EF/Model/Regions.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace ASC.Core.Common.EF.Model
+{
+    [Table("region")]
```
Comment (NikolayRechkin) [inline common/ASC.Core.Common/EF/Model/Regions.cs]: table name is "regions"

File: common/ASC.Feed/Data/FeedAggregateDataProvider.cs [hunk 4]
```diff
@@ -43,35 +47,37 @@ public class FeedAggregateDataProvider
...
         {
-            var db = GetDb();
-            db.ExecuteNonQuery(new SqlInsert("feed_last", true).InColumnValue("last_key", key).InColumnValue("last_date", value));
+            var feedLast = new FeedLast
+            {
+                LastKey = key,
+                LastDate = value
+            };
+
+            FeedDbContext.FeedLast.Add(feedLast);
```
Comment (NikolayRechkin) [inline common/ASC.Feed/Data/FeedAggregateDataProvider.cs]: need to use AddOrUpdate function

File: common/ASC.Feed/Data/FeedAggregateDataProvider.cs [hunk 5]
```diff
@@ -93,58 +99,61 @@ public void SaveFeeds(IEnumerable<FeedRow> feeds, string key, DateTime value)
...
                 foreach (var u in f.Users)
                 {
-                    i2.Values(f.Id, u.ToString());
+                    var feedUser = new FeedUsers
+                    {
+                        FeedId = f.Id,
+                        UserId = u
+                    };
+
+                    FeedDbContext.FeedUsers.Add(feedUser);
```
Comment (NikolayRechkin) [inline common/ASC.Feed/Data/FeedAggregateDataProvider.cs]: need to use AddOrUpdate function

File: common/ASC.Resource.Manager/ResourceData.cs [hunk 6]
```diff
@@ -25,160 +25,231 @@
...
 
-        public ResourceData(DbOptionsManager dbOptionsManager)
+        public ResourceData(DbContextManager<ResourceDbContext> dbContext)
         {
-            DbOptionsManager = dbOptionsManager;
+            DbContext = dbContext.Get(Dbid);
         }
 
-        public DateTime GetLastUpdate()
+        /* public DateTime GetLastUpdate()
```
Comment (NikolayRechkin) [inline common/ASC.Resource.Manager/ResourceData.cs]: remove comments

Comment (pavelbannov) [reply common/ASC.Resource.Manager/ResourceData.cs]: will be fixed in the next version

File: common/ASC.Core.Common/Data/DbTenantService.cs [hunk 7]
```diff
@@ -26,341 +26,351 @@
...
+                    Status = t.Status,
+                    StatusChanged = t.StatusChangeDate,
+                    PaymentId = t.PaymentId,
+                    LastModified = t.LastModified = DateTime.UtcNow,
+                    Industry = t.Industry,
+                    Spam = t.Spam,
+                    Calls = t.Calls
+                };
+
+                TenantDbContext.Tenants.Add(tenant);
```
Comment (NikolayRechkin) [inline common/ASC.Core.Common/Data/DbTenantService.cs]: need to add .Entity

**Pull request merged**